src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Eficent Business and IT Consulting Services S.L. 2019
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+import subprocess
+
+from .. import github
+from ..config import switchable
+from ..github import _git_call
+from ..manifest import _user_can_push
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            if not _user_can_push(gh, org, repo, username, ".", target_branch):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username} you are not allowed to rebase.\n\n"
+                    f"To do so you must either have push permissions on "
+                    f"the repository, or be a declared maintainer of all "
+                    f"modified addons.\n\n"
+                    f"If you wish to adopt an addon and become it's "
+                    f"[maintainer]"
+                    f"(https://odoo-community.org/page/maintainer-role), "
+                    f"open a pull request to add "
+                    f"your GitHub login to the `maintainers` key of its "
+                    f"manifest.",
+                )
+                return
+            # get remote name
+            remote = gh_pr.head._repo_owner
+            if not remote:
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username}, impossible to rebase "
+                    f"because unknown remote.",
+                )
+                return
+            # fetch pr and create local branch
+            current_branch = github.git_get_current_branch(cwd=".")
+            _git_call(
+                [
+                    "git",
+                    "fetch",
+                    "origin",
+                    f"refs/pull/{pr}/head:{remote}_{current_branch}_{pr}",
+                ]
+            )
```

Comment:
I may miss something obvious, but I don't see how this part can work unless cwd is a git repo. That's why I mentioned a needed call to temporary_clone before.